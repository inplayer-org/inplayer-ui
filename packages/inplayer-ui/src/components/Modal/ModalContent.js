// @flow
import React, { type Node } from 'packages/inplayer-ui/src/components/Modal/react';
import styled from 'styled-components';

type Props = {
  children: Node,
};

const ContentContainer = styled.div`
  padding: 0.625rem 1.25rem;
  box-sizing: border-box;
`;

const ModalContent = ({ children }: Props) => <ContentContainer>{children}</ContentContainer>;

export default ModalContent;
