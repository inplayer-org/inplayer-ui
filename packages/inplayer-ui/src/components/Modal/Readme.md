```jsx static
import { Modal } from 'inplayer-ui';

class App extends React.Component {
  this.state = {
    isModalOpen: false,
  };

  closeModal = () => {
    this.setState({
      isModalOpen: false,
    });
  };

  openModal = () => {
    this.setState({
      isModalOpen: true,
    });
  };

  render() {
    return (
      <div>
        <button onClick={this.openModal}>
          Open modal
        </button>

        <Modal
          title="Lorem Ipsum"
          isModalOpen={this.state.isModalOpen}
          closeModal={this.closeModal}
        >
           <p>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Phasellus
            pretium nec nisi pulvinar lacinia. Sed dapibus id dui vel varius.
            Duis erat ligula, posuere nec massa nec, ultricies bibendum velit.
            Nam vehicula lectus a convallis varius. Duis ullamcorper felis ac
            sem pulvinar pretium pellentesque id urna. Aliquam erat volutpat.
            Pellentesque sagittis congue lorem, at ultricies mauris lobortis a.
            Nam dignissim lectus eget tellus eleifend dictum.
          </p>
        </Modal>
      </div>
    );
  }
}
```

### Demo

```jsx
class App extends React.Component {
  // set initial state
  constructor() {
    super();

    this.state = {
      isModalOpen: false,
    };

    // bind functions
    this.closeModal = this.closeModal.bind(this);
    this.openModal = this.openModal.bind(this);
  }

  closeModal() {
    this.setState({
      isModalOpen: false,
    });
  }

  openModal() {
    this.setState({
      isModalOpen: true,
    });
  }

  render() {
    return (
      <div>
        <Button onClick={this.openModal}>Open modal</Button>

        <Modal
          title="Lorem Ipsum"
          isModalOpen={this.state.isModalOpen}
          closeModal={this.closeModal}
        >
          <p>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Phasellus
            pretium nec nisi pulvinar lacinia. Sed dapibus id dui vel varius.
            Duis erat ligula, posuere nec massa nec, ultricies bibendum velit.
            Nam vehicula lectus a convallis varius. Duis ullamcorper felis ac
            sem pulvinar pretium pellentesque id urna. Aliquam erat volutpat.
            Pellentesque sagittis congue lorem, at ultricies mauris lobortis a.
            Nam dignissim lectus eget tellus eleifend dictum.
          </p>
        </Modal>
      </div>
    );
  }
}

<App />;
```
