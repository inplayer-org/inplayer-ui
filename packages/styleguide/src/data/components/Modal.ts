import { ComponentData } from "../types";

const usage = `
  import { Modal } from 'inplayer-ui';

  class App extends React.Component {
    this.state = {
      isModalOpen: false,
    };

    closeModal = () => {
      this.setState({
        isModalOpen: false,
      });
    };

    openModal = () => {
      this.setState({
        isModalOpen: true,
      });
    };

    render() {
      return (
        <div>
          <button onClick={this.openModal}>
            Open modal
          </button>

          <Modal
            title="Lorem Ipsum"
            isModalOpen={this.state.isModalOpen}
            closeModal={this.closeModal}
          >
            <p>
              Lorem ipsum dolor sit amet, consectetur adipiscing elit. Phasellus
              pretium nec nisi pulvinar lacinia. Sed dapibus id dui vel varius.
              Duis erat ligula, posuere nec massa nec, ultricies bibendum velit.
              Nam vehicula lectus a convallis varius. Duis ullamcorper felis ac
              sem pulvinar pretium pellentesque id urna. Aliquam erat volutpat.
              Pellentesque sagittis congue lorem, at ultricies mauris lobortis a.
              Nam dignissim lectus eget tellus eleifend dictum.
            </p>
          </Modal>
        </div>
      );
    }
  }
`;

const demoCode = `class App extends React.Component {
  // set initial state
  constructor() {
    super();

    this.state = {
      isModalOpen: false,
    };

    // bind functions
    this.closeModal = this.closeModal.bind(this);
    this.openModal = this.openModal.bind(this);
  }

  closeModal() {
    this.setState({
      isModalOpen: false,
    });
  }

  openModal() {
    this.setState({
      isModalOpen: true,
    });
  }

  render() {
    return (
      <div>
        <Button onClick={this.openModal}>Open modal</Button>

        <Modal
          title="Lorem Ipsum"
          isModalOpen={this.state.isModalOpen}
          closeModal={this.closeModal}
        >
          <p>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Phasellus
            pretium nec nisi pulvinar lacinia. Sed dapibus id dui vel varius.
            Duis erat ligula, posuere nec massa nec, ultricies bibendum velit.
            Nam vehicula lectus a convallis varius. Duis ullamcorper felis ac
            sem pulvinar pretium pellentesque id urna. Aliquam erat volutpat.
            Pellentesque sagittis congue lorem, at ultricies mauris lobortis a.
            Nam dignissim lectus eget tellus eleifend dictum.
          </p>
        </Modal>
      </div>
    );
  }
}
`;

const Modal: ComponentData = {
  path: 'src/components/Modal/index.js',
  usage,
  examples: [
    {
      title: 'Demo',
      code: demoCode,
    }
  ],
  propsAndMethods: [
    { name: 'children', type: 'Node', default: 'Required' },
    { name: 'closeModal', type: 'function', default: 'Required' },
    { name: 'isModalOpen', type: 'boolean', default: 'Required' },
    { name: 'title', type: 'string', default: 'Required' },
    { name: 'className', type: 'string' },
    { name: 'style', type: 'Object', default: '{}' },
  ]
}

export default Modal;