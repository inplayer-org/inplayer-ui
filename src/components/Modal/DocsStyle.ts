import styled from 'styled-components';

export const ModalStyles = styled.div`
  display: grid;
  place-items: center;
  height: 400px;
`;
