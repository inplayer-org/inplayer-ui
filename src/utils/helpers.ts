import moment from 'moment';

export const getMonthOptions = (): Array<any> =>
  moment.months().map((label, value) => ({ value, displayName: label }));

export const getYearOptions = () => {
  const yearOptions: Array<any> = [];
  for (let i = moment().year() - 100; i <= moment().year(); i += 1) {
    yearOptions.push({ value: i.toString(), displayName: i.toString() });
  }

  return yearOptions;
};
