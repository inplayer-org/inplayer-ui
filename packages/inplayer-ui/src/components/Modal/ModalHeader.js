// @flow
import React, { type Node } from 'packages/inplayer-ui/src/components/Modal/react';
import styled from 'styled-components';
import colors from 'packages/inplayer-ui/src/config/colors';
import { uiColors } from 'packages/inplayer-ui/src/components/Modal/utils';
import { Typography } from 'packages/inplayer-ui/src/components/Modal/elements';

type Props = {
  children: Node,
  closeModal: () => any,
};

const HeaderWrapper = styled.div`
  padding: 1.25rem;
  box-sizing: border-box;
  position: relative;
  border-bottom: 1px solid ${colors.gray};
`;

const Title = styled(Typography)`
  margin: 0;
`;

const IconClose = styled.span`
  position: absolute;
  height: 1.125rem;
  width: 1.125rem;
  top: 0.625rem;
  right: 0.625rem;
  cursor: pointer;

  &::before {
    content: 'âœ•';
    color: ${uiColors('secondary.main')};
  }
`;

const ModalHeader = ({ children, closeModal }: Props) => (
  <HeaderWrapper>
    <Title variant="h4">{children}</Title>
    <IconClose onClick={closeModal} />
  </HeaderWrapper>
);

export default ModalHeader;
