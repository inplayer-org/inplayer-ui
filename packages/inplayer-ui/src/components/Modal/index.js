// @flow
import React, { type Node } from 'react';
import styled from 'styled-components';
import { ifProp } from 'styled-tools';
import colors from 'config/colors';
import { transparentize } from 'polished';
import ModalHeader from './ModalHeader';
import ModalContent from './ModalContent';

type Props = {
  isModalOpen: boolean,
  closeModal: () => any,
  children: Node,
  title: string,
  className?: string,
  style?: Object,
};

const ModalWrapper = styled.div`
  display: ${ifProp('isOpen', 'flex', 'none')};
  height: 100%;
  width: 100%;
  position: fixed;
  z-index: 100;
  top: 0;
  left: 0;
  background: ${transparentize(0.28, colors.black)};
`;

const ModalContainer = styled.div`
  background: ${colors.white};
  width: 72%;
  max-width: 980px;
  border-radius: 3px;
  margin: auto;
`;

const Modal = ({ isModalOpen, closeModal, children, title, className, style }: Props) => (
  <ModalWrapper isOpen={isModalOpen} onClick={isModalOpen ? closeModal : undefined}>
    <ModalContainer className={className} style={style} onClick={e => e.stopPropagation()}>
      <ModalHeader closeModal={closeModal}>{title}</ModalHeader>
      <ModalContent>{children}</ModalContent>
    </ModalContainer>
  </ModalWrapper>
);

Modal.defaultProps = {
  className: '',
  style: {},
};

export default Modal;
