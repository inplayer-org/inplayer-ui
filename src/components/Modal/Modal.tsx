import React, { ReactChild } from 'react';
import styled from 'styled-components';
import { ifProp } from 'styled-tools';
import { transparentize } from 'polished';

import colors from '../../theme/colors';
import ModalHeader from './ModalHeader';
import ModalContent from './ModalContent';

interface ModalWrapperProps {
  isOpen: boolean;
}

const ModalWrapper = styled.div<ModalWrapperProps>`
  display: ${ifProp('isOpen', 'flex', 'none')};
  height: 100%;
  width: 100%;
  position: fixed;
  z-index: 100;
  top: 0;
  left: 0;
  background: ${transparentize(0.28, colors.black)};
  overflow-y: scroll;
`;

const ModalContainer = styled.div<{ isOverflowVisible?: boolean }>`
  background: ${({ theme }) => theme.palette.background.main};
  color: ${({ theme }) => theme.palette.text.main};
  width: 72%;
  max-width: 980px;
  border-radius: 3px;
  margin: auto;
  overflow: ${({ isOverflowVisible }) => (isOverflowVisible ? 'visible;' : 'hidden;')};
`;

type Props = {
  isModalOpen: boolean;
  closeModal: () => any;
  title: string;
  isOverflowVisible?: boolean;
  className?: string;
  children: ReactChild;
};

const Modal = ({
  isOverflowVisible = false,
  isModalOpen,
  closeModal,
  children,
  title,
  className = '',
}: Props) => (
  <ModalWrapper
    className={className}
    isOpen={isModalOpen}
    onClick={isModalOpen ? closeModal : undefined}
  >
    <ModalContainer isOverflowVisible={isOverflowVisible} onClick={(e) => e.stopPropagation()}>
      <ModalHeader closeModal={closeModal}>{title}</ModalHeader>
      <ModalContent>{children}</ModalContent>
    </ModalContainer>
  </ModalWrapper>
);

export default Modal;
