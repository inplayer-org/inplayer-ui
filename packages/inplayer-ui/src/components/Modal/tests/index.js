import { generateSnapshots } from 'packages/inplayer-ui/src/components/Modal/tests/helpers';

import Modal from '../index';

generateSnapshots(Modal);
