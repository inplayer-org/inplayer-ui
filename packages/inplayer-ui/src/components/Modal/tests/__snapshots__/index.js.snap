// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal renders correctly 1`] = `
.c4 {
  font-family: Roboto;
  font-weight: 100;
  color: #286090;
  font-size: 1.75rem;
}

.c2 {
  padding: 1.25rem;
  box-sizing: border-box;
  position: relative;
  border-bottom: 1px solid #d7dde5;
}

.c3 {
  margin: 0;
}

.c5 {
  position: absolute;
  height: 1.125rem;
  width: 1.125rem;
  top: 0.625rem;
  right: 0.625rem;
  cursor: pointer;
}

.c5::before {
  content: 'âœ•';
  color: #286090;
}

.c6 {
  padding: 0.625rem 1.25rem;
  box-sizing: border-box;
}

.c0 {
  display: none;
  height: 100%;
  width: 100%;
  position: fixed;
  z-index: 100;
  top: 0;
  left: 0;
  background: rgba(0,0,0,0.72);
}

.c1 {
  background: #fff;
  width: 72%;
  max-width: 980px;
  border-radius: 3px;
  margin: auto;
}

<div
  className="c0"
>
  <div
    className="c1"
    onClick={[Function]}
    style={Object {}}
  >
    <div
      className="c2"
    >
      <h4
        className="c3 c4"
        color=""
        style={Object {}}
      />
      <span
        className="c5"
      />
    </div>
    <div
      className="c6"
    />
  </div>
</div>
`;
